server/src/test/java/org/apache/druid/server/coordinator/duty/KillUnusedSegmentsTest.java

```diff
@@ -225,6 +225,154 @@ public void testKillWithMultipleDatasources()
     validateLastKillStateAndReset(DS2, null);
   }
 
+  /**
+   * Set up multiple datasources {@link #DS1}, {@link #DS2} and {@link #DS3} with unused segments with 2 kill task
+   * slots. Running the kill duty each time should pick at least one unique datasource in a round-robin manner.
+   */
+  @Test
+  public void testKillMultipleDatasourcesInRoundRobinManner()
```

Comment:
Super Nit: Can we avoid using the suffix `Manner`?
Maybe rename to `testRoundRobinKillMultipleDatasources()`?
Same comment for the other test name.